### PR TITLE
Ewl-974: new login modal

### PR DESCRIPTION
**Pull request by ra-phael**
# About the type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Test plan

Mostly snapshots

# Checklist
### Before code review

- [ ] Move Jira ticket into PR
- [x] Check with another developers that features sounds good
- [x] Add PropTypes
- [x] Figure out best way to handle background images

### After code review

- [ ] Move Jira ticket into QA

### After QA validation
If conflicts are present in github:
  - [ ] `git pull --rebase origin master`
  - [ ] Fix conflicts
  - [ ] `git rebase --continue` repeat until there is no more conflicts
  - [ ] `git push --force-with-lease`

If there's no conflicts
- [ ] squash and merge
- [ ] delete branch

### CODE REVIEW

File: src/images.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  background-image: url(${retinaImageSrc.default});
+  @media (-webkit-min-device-pixel-ratio: 2) {
+    background-image: url(${retinaImageSrc.x2});
+  }
+  @media (-webkit-min-device-pixel-ratio: 3) {
+    background-image: url(${retinaImageSrc.x3});
+  }
+`;
+
+export const responsiveRetinaImage = image => css`
```
Comment (ra-phael) [inline src/images.js]: Took the content of that file & tests from the watch eurosport landing project, but changed the breakpoints to match the ones we have

File: src/components/Input/input.component.js [hunk 2]
```diff
@@ -20,32 +20,45 @@ const Input = ({ placeholder, type, value, onChange, textButton }) => {
...
+      </MainContainer>
+    </>
   );
 };
 
 Input.defaultProps = {
   placeholder: '',
   type: INPUT_TEXT,
   textButton: '',
+  hasLabelInside: true,
```
Comment (ra-phael) [inline src/components/Input/input.component.js]: Quite a few changes to the Input component but at least setting that to true by default make them non breaking changes...

File: src/modules/Login/index.js [hunk 3]
```diff
@@ -1,11 +1,12 @@
 import React, { useState, useCallback } from 'react';
-import styled from 'react-emotion';
+import styled, { css } from 'react-emotion';
 import PropTypes from 'prop-types';
 import Input from 'src/components/Input/input.component';
 import ErrorBanner from 'src/elements/ErrorBanner';
 import Button from 'src/elements/Button';
 import ArrowLink from 'src/elements/ArrowLink';
 import UnderlineLink from 'src/elements/UnderlineLink';
+import * as breakpoints from '../../breakpoints';
```
Comment (JulienKode) [inline src/modules/Login/index.js]: Should we have an absolute import ?

File: src/modules/LoginModal/index.test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  it('renders Login Modal as expected', () => {
+    const props = createProps();
+    const wrapper = mount(
+      <ThemeProvider theme={theme}>
+        <LoginModal {...props} />
+      </ThemeProvider>
+    );
+    // Expect
+    expect(wrapper).toMatchSnapshot();
```
Comment (JulienKode) [inline src/modules/LoginModal/index.test.js]: Can we avoid doing a snapshot of the whole theme ? 
each time the theme will change we will have to update this test 

Then as we test the LoginModal component we can expect something like that:
`expect(wrapper.find(LoginModal)).toMatchSnapshot();`

File: src/modules/LoginModal/index.test.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { ThemeProvider } from 'emotion-theming';
+import { mount } from 'enzyme';
+import theme from '../../theme';
```
Comment (JulienKode) [inline src/modules/LoginModal/index.test.js]: Absolute import

File: src/modules/LoginModal/index.js [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+import PropTypes from 'prop-types';
+
+import CheckIcon from 'src/assets/red-check.svg';
+import ESPlayerLogo from '../../assets/eurosport-player-logo.component.svg';
+
+import * as breakpoints from '../../breakpoints';
+import { coreLightMinus1, coreNeutral9, suitGray } from '../../colors';
+import Login from '../Login';
+import Button from '../../elements/Button';
+import { H2 } from '../../typography';
```
Comment (JulienKode) [inline src/modules/LoginModal/index.js]: Can we use the `Theme` from the provider ?

File: src/modules/LoginModal/index.js [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+const StyledSubscribeButton = styled(Button)`
+  max-width: 222px;
+`;
+
+const Separator = styled.div`
+  border-bottom: solid 2px ${coreNeutral9};
+  margin: 30px 0;
+  ${breakpoints.large(css`
+    border-right: solid 2px ${coreNeutral9};
```
Comment (JulienKode) [inline src/modules/LoginModal/index.js]: Should we use colors from theme ?

File: src/modules/LoginModal/index.test.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    subscribeUrl: 'es.com/subscribe',
+    subscribeText: 'Subscribe',
+    needHelpText: 'Need Help?',
+    needHelpUrl: 'es.com/help',
+    callToActionText: 'New to eurosport player?',
+    passwordPlaceholder: 'Password',
+    emailPlaceholder: 'Email',
+    signInText: 'SignIn',
+    showPasswordText: 'Show',
+    errorMessage: undefined,
```
Comment (JulienKode) [inline src/modules/LoginModal/index.test.js]: Should we have `null` instead ?

File: src/modules/LoginModal/__snapshots__/index.test.js.snap [hunk 9]
```diff
@@ -0,0 +1,914 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Login Modal renders Login Modal as expected 1`] = `
+<ThemeProvider
+  theme={
```
Comment (JulienKode) [inline src/modules/LoginModal/__snapshots__/index.test.js.snap]: This part contains the whole theme

File: src/modules/LoginModal/index.js [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+  color: ${coreLightMinus1};
+  padding-left: 30px;
+  font-size: 16px;
+  margin-bottom: 8px;
+  line-height: 1.5;
+  list-style: none;
+  background: url(${CheckIcon}) left 5px no-repeat;
+`;
+
+const MarketingContent = ({ title, subtitle, checklist }) => (
```
Comment (JulienKode) [inline src/modules/LoginModal/index.js]: Should we split that file into multiples components ?

File: src/modules/LoginModal/index.js [hunk 11]
```diff
@@ -0,0 +1,169 @@
+import React, { useState } from 'react';
+import styled, { css } from 'react-emotion';
+import PropTypes from 'prop-types';
+
+import CheckIcon from 'src/assets/red-check.svg';
+import ESPlayerLogo from '../../assets/eurosport-player-logo.component.svg';
```
Comment (JulienKode) [inline src/modules/LoginModal/index.js]: absolute import here too

File: src/modules/LoginModal/index.js [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+      <MobileHeader>
+        <StyledESPlayerLogo />
+      </MobileHeader>
+      {isLoginScreen ? (
+        <Login {...loginProps} />
+      ) : (
+        <>
+          <MarketingContent {...marketingMessages} />
+          <Button arrowType="arrow">Subscribe Now</Button>
+          <SignInButton type="form" arrowType="arrow" onClick={() => setIsLoginScreen(prevScreen => !prevScreen)}>
```
Comment (JulienKode) [inline src/modules/LoginModal/index.js]: Can we have that into a `useCallback` ? Then we do not have to re-create the function over and over again

File: src/modules/VideoPlayerModal/index.js [hunk 13]
```diff
@@ -1,35 +1,77 @@
...
+    top: 67px;
+    right: 38px;
+  `)};
+  ${breakpoints.large(css`
+    top: 43px;
+    right: 38px;
+  `)};
+
+  path {
+    fill: ${coreLightMinus1};
```
Comment (JulienKode) [inline src/modules/VideoPlayerModal/index.js]: Can we use it from the theme ?

File: src/modules/VideoPlayerModal/index.stories.js [hunk 14]
```diff
@@ -18,10 +19,28 @@ const createDefaultProps = () => ({
...
+    ],
+  },
 });
 
-storiesOf('Modules|VideoPlayerModal', module).add('with Login', () => (
-  <VideoPlayerModal>
-    <Login {...createDefaultProps()} />
-  </VideoPlayerModal>
-));
+// eslint-disable-next-line no-console
```
Comment (JulienKode) [inline src/modules/VideoPlayerModal/index.stories.js]: Can we use https://www.npmjs.com/package/@storybook/addon-actions for this event ?

Comment (ra-phael) [reply src/modules/VideoPlayerModal/index.stories.js]: I didn't know about that, really cool!

File: src/modules/LoginModal/index.js (hunk 7 above)
Comment (ra-phael) [reply src/modules/LoginModal/index.js]: For?

File: src/modules/LoginModal/index.js (hunk 6 above)
Comment (ra-phael) [reply src/modules/LoginModal/index.js]: For?

File: src/modules/LoginModal/index.test.js (hunk 8 above)
Comment (ra-phael) [reply src/modules/LoginModal/index.test.js]: Why not... It's undefined in its default props anyway :)

File: src/modules/LoginModal/index.js (hunk 12 above)
Comment (ra-phael) [reply src/modules/LoginModal/index.js]: Good point!

File: src/modules/LoginModal/index.js (hunk 6 above)
Comment (ra-phael) [reply src/modules/LoginModal/index.js]: Color is now coming from the theme

File: src/modules/LoginModal/index.js (hunk 10 above)
Comment (ra-phael) [reply src/modules/LoginModal/index.js]: If the need to use this component elsewhere arises I think it'll be quite easy for anyone to extract it from here.

File: src/modules/VideoPlayerModal/index.js (hunk 13 above)
Comment (ra-phael) [reply src/modules/VideoPlayerModal/index.js]: done :)

File: src/modules/LoginModal/index.js (hunk 7 above)
Comment (ra-phael) [reply src/modules/LoginModal/index.js]: Not sure about this one, it adds complexity as we can't directly use theme within `css` and little benefit

File: src/modules/LoginModal/index.js (hunk 10 above)
Comment (JulienKode) [reply src/modules/LoginModal/index.js]: Makes sense

File: src/components/Input/input.component.js (hunk 2 above)
Comment (davidgarry) [reply src/components/Input/input.component.js]: I should be ok with that, but as this input component is only used in the login modal, I think it's better to set it to false by default. What do you think ?

File: src/modules/Login/index.js [hunk 15]
```diff
@@ -46,34 +48,41 @@ const Login = ({
...
-      </ComponentContainer>
-      <ComponentContainer>
-        <Button type="form" onClick={onSignInSubmit}>
-          {signInText}
-        </Button>
-        <InvisibleSubmit type="submit" />
-      </ComponentContainer>
+      <InputsContainer>
+        <ComponentContainer>
+          <Input value={email} onChange={onEmailChange} placeholder={emailPlaceholder} hasLabelInside={false} />
```
Comment (davidgarry) [inline src/modules/Login/index.js]: See my previous comment, if it's set by default at false, you can remove the `hasLabelInside` prop.

File: src/modules/Login/index.js [hunk 16]
```diff
@@ -46,34 +48,41 @@ const Login = ({
...
+        </ComponentContainer>
+        <ComponentContainer>
+          <Input
+            value={password}
+            onChange={onPasswordChange}
+            placeholder={passwordPlaceholder}
+            type="password"
+            textButton={showPasswordText}
+            hasLabelInside={false}
```
Comment (davidgarry) [inline src/modules/Login/index.js]: Same as above

File: src/modules/Login/index.test.js [hunk 17]
```diff
@@ -39,6 +39,14 @@ describe('Login', () => {
       const props = createProps({ errorMessage: 'Impossible to connect' });
       const wrapper = shallow(<Login {...props} />);
 
+      // Expect
+      expect(wrapper).toMatchSnapshot();
```
Comment (davidgarry) [inline src/modules/Login/index.test.js]: We have two snapshots here. Can you remove the second one and just test that the subscription section is not rendered ?

**Comment by ra-phael** [comment]
Thanks for your feedback @davidgarry ! I just made the requested changes

File: src/modules/LoginModal/index.test.js (hunk 8 above)
Comment (JulienKode) [reply src/modules/LoginModal/index.test.js]: Yes it's just philosophical, not mandatory 😊

File: src/modules/LoginModal/index.js (hunk 6 above)
Comment (JulienKode) [reply src/modules/LoginModal/index.js]: Perfect 😊

File: src/modules/LoginModal/index.js (hunk 7 above)
Comment (JulienKode) [reply src/modules/LoginModal/index.js]: Yes the only way to access to the theme in css is a function

`
${({ theme }) => breakpoints.large(css`
    border-right: solid 2px ${theme.colors.stuff};
`)}
`

File: src/modules/LoginModal/index.stories.js [hunk 18]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { text } from '@storybook/addon-knobs';
+import LoginModal from '.';
+
+const createDefaultProps = () => ({
+  onSubmit: () => null,
```
Comment (JulienKode) [inline src/modules/LoginModal/index.stories.js]: Should we use ? https://github.com/storybookjs/storybook/tree/master/addons/actions

**Comment by JulienKode** [comment]
@ra-phael can we update to use the latest ETP version ? 🤔

File: src/modules/VideoPlayerModal/index.js [hunk 19]
```diff
@@ -1,35 +1,76 @@
...
-import styled from 'react-emotion';
+import styled, { css } from 'react-emotion';
 import PropTypes from 'prop-types';
 import * as breakpoints from 'src/breakpoints';
+import CloseIcon from '../../assets/close-cross.component.svg';
+import ESPlayerBG from '../../assets/backgrounds/eurosport-player';
+import { responsiveRetinaImage } from '../../images';
 
-const VideoPlayerModal = ({ children }) => (
+const VideoPlayerModal = ({ children, onClose }) => (
```
Comment (JulienKode) [inline src/modules/VideoPlayerModal/index.js]: We may have to update JWPLayerETP/index.js with 

```
const LoginModal = ({ onClose, ...props }) => (
  <VideoPlayerModal onClose={onClose}>
    <Login {...props} />
  </VideoPlayerModal>
);
```

Comment (JulienKode) [reply src/modules/VideoPlayerModal/index.js]: As `LoginModal` received props from ETP

File: src/modules/LoginModal/index.stories.js (hunk 18 above)
Comment (ra-phael) [reply src/modules/LoginModal/index.stories.js]: Done :)

**Pull request merged**